quadriga/src/main/java/edu/asu/diging/quadriga/web/DisplayCollectionController.java

```diff
@@ -0,0 +1,112 @@
+    @Autowired
+    private PredicateManager predicateManager;
+    
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    @RequestMapping(value = "/auth/collections/{id}", method = RequestMethod.GET)
+    public String get(@PathVariable String id, Model model) {
+        
+        Collection collection;
+        try {
+            collection = collectionManager.findCollection(id);
+            if(collection == null) {
+                return "error404Page";
+            }
+        } catch (InvalidObjectIdException e) {
+            return "error404Page";
+        }
+        
+        ObjectId collectionId = collection.getId();
+        
+        List<EventGraph> eventGraphs = eventGraphService.findAllEventGraphsByCollectionId(collectionId);
+        
+        if(!eventGraphs.isEmpty()) {
+            // We get last network submission info by getting the last EventGraph which will be a part of the last network
+            EventGraph lastNetwork = eventGraphs.get(0);
+            model.addAttribute("lastNetworkSubmittedAt", lastNetwork.getCreationTime().atZoneSameInstant(ZoneId.systemDefault()));
+            model.addAttribute("lastNetworkSubmittedBy", lastNetwork.getSubmittingApp());
+        }
+        
+        model.addAttribute("collectionName", collection.getName());
+        model.addAttribute("description", collection.getDescription());
+        model.addAttribute("creationTime", collection.getCreationTime().atZoneSameInstant(ZoneId.systemDefault()));
+        
+        // One network may have multiple eventGraphs, but all of them will have same sourceURI in the context
+        // This sourceURI will be used to group eventGraphs together that belong to the same network
+        model.addAttribute("numberOfSubmittedNetworks", eventGraphs.stream().collect(Collectors.groupingBy(event -> event.getContext().getSourceUri())).size());
+        
+        // Get default mappings from Concepts
+        model.addAttribute("defaultMappings", getNumberOfDefaultMappings(collectionId.toString()));
+        
+        return "auth/displayCollection";
+        
+    }
+    
+    /**
+     * This method returns the number of default mappings present in the collection
+     * One MappedTripleGroup will exist for the "DefaultMappings" for this collection
+     * To get this number of default mappings, this method will check how many 'Predicates' have
+     * this mappedTripleGroupId linked to them
+     * This is because every default mapping has one predicate
+     * So, if the MappedTripleGroupId is present on n predicates, this collection
+     * must have n defaultMappings 
+     * 
+     * @param collectionId used to find mappedTripleGroupId
+     * @return the number of default mappings
+     */
+    private int getNumberOfDefaultMappings(String collectionId) {
+        try {
+            MappedTripleGroup mappedTripleGroup = mappedTripleGroupService.findByCollectionIdAndMappingType(collectionId, MappedTripleType.DEFAULT_MAPPING);
```

Comment:
same as above, should be done via db

Reply:
@jdamerow , 
 This method is fetching mappedTripleGroups directly from mongo db, followed by fetching corresponding predicates from neo4j to get the total number of default mappings. Please clarify what is the expected change here?

Reply:
this probably refers to the way the number of predicates is retrieved. Getting the triple group then counting the predicates. Instead there should be a call to the db to get the number of predicates (so it doesns't load alll triples into memory).